lib/getPackageDetails.js

```diff
@@ -70,6 +72,11 @@ module.exports = function getPackageDetails(
     return Promise.resolve(null);
   }
   const key = `${name}@${versionLoose}`;
+  const scope = (/(@[^\/]+)\/.*/.test(name) && /(@[^\/]+)\/.*/.exec(name)[1]) || ``;
```

Comment:
I suggest to use https://github.com/npm/npm-package-arg (because `npm` uses it)
or alternatively copy logic from it to be on safe side 😉

```js
this.scope = name[0] === '@' ? name.slice(0, name.indexOf('/')) : undefined
// scoped packages in couch must have slash url-encoded, e.g. @foo%2Fbar
this.escapedName = name.replace('/', '%2f')

```

It's difficult to say if your regexp works exactly. the same.

Reply:
just changed the parsing to the same logic, thanks for the hint